contracts/token-distribution/SaleRounds.sol

```diff
@@ -204,115 +159,186 @@ contract SaleRounds is TokenDistribution, GameOwner, ERC20 {
         }
 
         reserveTokensInternal(roundType, _to, _amount);
+        emit ReserveTokensEvent(_roundType, _amount, _to);
     }
 
     // @_amount is going be decimals() == default(18) digits
-    function mintTokensForPublic(string calldata _roundType, address _to, uint _amount) public
-        onlyOwner isRoundActive(_roundType) {
+    function mintTokensForPublic(string calldata _roundType, address _to, uint _amount) external
+    onlyOwner {
         RoundType roundType = getRoundTypeByKey(_roundType);
 
         require(roundType == RoundType.PUBLIC , "round type is not valid");
-        require(roundDistribution[roundType].totalRemaining >= _amount, "total remaining amount is not enough" );
+        require(roundDistribution[roundType].totalRemaining >= _amount, "total remaining amount is not enough");
 
         roundDistribution[roundType].totalRemaining -= _amount;
         _mint(_to, _amount);
+        tokensClaimable = true;
     }
 
-    function claimTokens(string calldata _roundType, address _to) public
-        isRoundActive(_roundType) claimableRound(_roundType) {
+    function claimTokens(string calldata _roundType, address _to) external
+    claimableRound(_roundType) {
         RoundType roundType = getRoundTypeByKey(_roundType);
+        require(tokensClaimable, "Tokens are still not claimable");
         require(_msgSender() == _to, "Sender is not a recipient");
 
         ClaimInfo memory claimInfo = ClaimInfo({cliff: roundDistribution[roundType].cliff,
-                                                vesting: roundDistribution[roundType].vesting,
-                                                balance: reservedBalances[roundType][_to],
-                                                claimedBalance: claimedBalances[roundType][_to],
-                                                //granularity can be 30 * 24 * 60 * 60 for monthly or 24 * 60 * 60 for daily? add a field and make constants
-                                                periodGranularity: roundDistribution[roundType].vestingGranularity, //e.g. Days or Months (in seconds!)
-                                                startTime: roundDistribution[roundType].startTime,
-                                                secondsVested: 0,
-                                                vestingForUserPerSecond: 0
-                                            });
+        vesting: roundDistribution[roundType].vesting,
+        balance: reservedBalances[roundType][_to],
+        claimedBalance: claimedBalances[roundType][_to],
+        //granularity can be 30 * 24 * 60 * 60 for monthly or 24 * 60 * 60 for daily? add a field and make constants
```

Comment:
this comment is vestigial, please remove. This came from some pseudocode I shared with veysel.